ibm/data_source_ibm_scc_account_locations.go

```diff
@@ -0,0 +1,165 @@
+// Licensed under the Mozilla Public License v2.0
+
+package ibm
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/IBM/scc-go-sdk/adminserviceapiv1"
+)
+
+func dataSourceIbmSccAccountLocations() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceIbmSccAccountLocationsRead,
+
+		Schema: map[string]*schema.Schema{
+			"locations": &schema.Schema{
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The programatic ID of the location that you want to work in.",
+						},
+						"main_endpoint_url": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The base URL for the service.",
+						},
+						"governance_endpoint_url": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The endpoint that is used to call the Configuration Governance APIs.",
+						},
+						"results_endpoint_url": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The endpoint that is used to get the results for the Configuration Governance component.",
+						},
+						"compliance_endpoint_url": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The endpoint that is used to call the Posture Management APIs.",
+						},
+						"analytics_endpoint_url": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The endpoint that is used to generate analytics for the Posture Management component.",
+						},
+						"si_endpoint_url": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The endpoint that is used to call the Security Insights APIs.",
+						},
+						"regions": &schema.Schema{
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"id": &schema.Schema{
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "The programatic ID of the available regions.",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceIbmSccAccountLocationsRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	adminServiceApiClient, err := meta.(ClientSession).AdminServiceApiV1()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	listLocationsOptions := &adminserviceapiv1.ListLocationsOptions{}
+
+	locations, response, err := adminServiceApiClient.ListLocationsWithContext(context, listLocationsOptions)
```

Comment:
I think add supports for pagination to list all pages and give the complete list out

Reply:
This might not be needed as pagination is not necessary for this specific returned payload.